### PR TITLE
Adds data prepper decompress processor documentation

### LINKED ISSUE #6407: [DOC] Data Prepper Decompress Processor
**What do you want to do?**
 
- [ ] Request a change to existing documentation
- [x] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** Provide a summary of the request and all versions that are affected.

The decompress compressor is a new processor for Data Prepper (https://github.com/opensearch-project/data-prepper/issues/4016)

### PR DESCRIPTION
**Pull request by graytaylor0**
### Description
Adds documentations for data prepper's new `decompress` processor.

### Issues Resolved
Closes [#6407 ](https://github.com/opensearch-project/documentation-website/issues/6407)

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by Naarcha-AWS — APPROVED** [review]

**Review by dlvenable — COMMENTED** [review]
@graytaylor0 , @Naarcha-AWS , I have a couple of small comments. But, other than that I'm good to proceed.

File: _data-prepper/pipelines/configuration/processors/decompress.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+layout: default
+title: decompress
+parent: Processors
+grand_parent: Pipelines
+nav_order: 40
+---
+
+# decompress
+
+The `decompress` processor decompresses the `base64`-encoded compressed fields inside of an event.
```
Comment (Naarcha-AWS) [inline _data-prepper/pipelines/configuration/processors/decompress.md:11]:
```suggestion
The `decompress` processor decompresses any base64-encoded compressed fields inside of an event.
```

File: _data-prepper/pipelines/configuration/processors/decompress.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+layout: default
+title: decompress
+parent: Processors
+grand_parent: Pipelines
+nav_order: 40
+---
+
+# decompress
+
+The `decompress` processor decompresses any base64-encoded compressed fields inside of an event.
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/processors/decompress.md:11]:
```suggestion
The `decompress` processor decompresses any Base64-encoded compressed fields inside of an event.
```

File: _data-prepper/pipelines/configuration/processors/decompress.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+## Configuration
+
+Option | Required | Type | Description
+:--- | :--- | :--- | :---
+`keys` | Yes | List<String> | The fields in the event that will be decompressed.                                                                                          
+`type` | Yes | Enum | The type of decompression to use for the `keys` in the event. Only `gzip` is supported.                                           
+`decompress_when` | No | String| A [Data Prepper conditional expression](https://opensearch.org/docs/latest/data-prepper/pipelines/expression-syntax/) that determines when the `decompress` processor will run on certain events.
+`tags_on_failure` | No | List<String> | A list of strings to tag events with when the processor fails to decompress the `keys` inside the event. Defaults to `_decompression_failure`.
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/processors/decompress.md:20]:
```suggestion
`tags_on_failure` | No | List<String> | A list of strings with which to tag events when the processor fails to decompress the `keys` inside an event. Defaults to `_decompression_failure`.                               
```

File: _data-prepper/pipelines/configuration/processors/decompress.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+Option | Required | Type | Description
+:--- | :--- | :--- | :---
+`keys` | Yes | List<String> | The fields in the event that will be decompressed.                                                                                          
+`type` | Yes | Enum | The type of decompression to use for the `keys` in the event. Only `gzip` is supported.                                           
+`decompress_when` | No | String| A [Data Prepper conditional expression](https://opensearch.org/docs/latest/data-prepper/pipelines/expression-syntax/) that determines when the `decompress` processor will run on certain events.
+`tags_on_failure` | No | List<String> | A list of strings to tag events with when the processor fails to decompress the `keys` inside the event. Defaults to `_decompression_failure`.                               
+
+## Usage
+
+The following example shows the `decompress` processor when used in `pipelines.yaml`:
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/processors/decompress.md:24]:
```suggestion
The following example shows the `decompress` processor used in `pipelines.yaml`:
```

File: _data-prepper/pipelines/configuration/processors/decompress.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+processor:
+  - decompress:
+      decompress_when: '/some_key == null'
+      keys: [ "base_64_gzip_key" ]
+      type: gzip
+```
+
+## Metrics 
+
+The following table describes common [Abstract processor](https://github.com/opensearch-project/data-prepper/blob/main/data-prepper-api/src/main/java/org/opensearch/dataprepper/model/processor/AbstractProcessor.java) metrics.
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/processors/decompress.md:36]: Should the linked text be either "abstract processor" or "AbstractProcessor"?

**Review by natebower — COMMENTED** [review]
@graytaylor0 @Naarcha-AWS Just a few minor changes. Thanks!

File: _data-prepper/pipelines/configuration/processors/decompress.md (hunk 5 above)
Comment (Naarcha-AWS) [inline _data-prepper/pipelines/configuration/processors/decompress.md:36]:
```suggestion
The following table describes common [abstract processor](https://github.com/opensearch-project/data-prepper/blob/main/data-prepper-api/src/main/java/org/opensearch/dataprepper/model/processor/AbstractProcessor.java) metrics.
```

**Pull request merged**